src/components/AssetsComponent.jsx

```diff
@@ -20,46 +21,66 @@ export default class AssetsComponent extends Component {
   };
 
   componentDidMount() {
-    const { activePage, assetsList } = this.props;
-
-    const assetsEmpty = isEmpty(assetsList);
-
     // TODO: fix the logic so that assets are fetched when you create an asset before fetching
     // assets, otherwise, you'll only display 1 row in assets table yet there are more than one
     // assets
-    if (assetsEmpty) {
-      this.props.getAssetsAction(activePage, this.state.limit);
-    }
+    this.loadAssets();
 
     this.props.loadAllAssetModels();
     this.props.loadDropdownAssetTypes();
   }
 
+  loadAssets = () => {
+    const { activePage, assetsList, match } = this.props;
+    const { assetStatus } = match.params;
+
+    const assetsEmpty = isEmpty(assetsList);
+    const assetStat = assetStatus === undefined ? '' : assetStatus;
+    const shouldReload = this.props.assetStatus !== assetStat;

```

Comment:
Please relocate `shouldReload` and `assetsEmpty` to the `AssetsContainer.jsx`. Remember that `mapStateToProps` takes two parameters. I.e. `state` and `ownProps`. Therefore, you will have something like:
```jsx
const mapStateToProps = ({ assets }, ownProps) => {
  const { params } = ownProps.match;

  // NOTE: undefined is falsy. Therefore a string will never be equal to undefined, which means that `assetStat` is unnecessary
  const shouldReload = params.status !== assets.status;
  const assetsEmpty = isEmpty(assets.assetsList);

  return {
    shouldFetchAssets: shouldReload || assetsEmpty,
    ...
  };
}